### DISCUSSION TITLE
Eco 7/Cheap overnight electric

### DISCUSSION THREAD

**Discussion by GeneralSprint** [jasonacox/Powerwall-Dashboard#370] [Ideas]
I don't know if this has been proposed before, or if this exists outside of the UK, but many UK energy suppliers offer cheaper rates overnight. Between 4:30am and 12:30am, (day time), I pay 29p per kWh. But between 12:30 and 4:30am (overnight), I only pay 9p per kWh. My Powerwall charges overnight using this cheap grid electricity, and avoids using grid power all day till the battery runs out (which in the winter it often does).

As it stands I can only input a single cost per kWh. Would it be possible to set these variables, and have them factored into the usage/cost calculator? 

Thanks in advance
Rich

**Comment by youzer-name** [comment]
The basic dashboards don't have any system for capturing time of use pricing. Currently, the Powerwall Dashboard system doesn't store historical prices. The cost/savings calculations are all done as the data is queried from the database.

There are some options that have been discussed before. You might want to search the discussions for 'TOU' or 'time of use' to see more, but here are links to two:

1. This discussion gets into how it might be possible to implement time of use pricing using Flux queries: https://github.com/jasonacox/Powerwall-Dashboard/discussions/225

2. This discussion documents what I have set up: https://github.com/jasonacox/Powerwall-Dashboard/discussions/82. It's also been added to the project documents here: https://github.com/jasonacox/Powerwall-Dashboard/tree/main/tools/mysql

This method involves setting up a separate database engine (MySQL) and copying data into it from Influx (which happens automatically using a Python script). As the data is inserted into MySQL, the cost/savings calculations are done and the costs stored for each row, meaning that as time goes on and costs change, the historical queries will return the correct costs for the past dates. It does introduce a number of additional moving parts to the system, but once it's set up, it pretty much runs on autopilot. 

I also posted some dashboards that were built off of the time of use pricing here: https://github.com/jasonacox/Powerwall-Dashboard/discussions/119. I need to update that post, as I've since had to correct some issues in the panels in my Powerwall History dashboard.

I run a customized setup and don't use the upgrade scripts when changes are made. If you're starting from a stock installation you should be able to add additional dashboards alongside the stock ones in Grafana without breaking the upgrade process.
